ui/components/app/transaction-list-item/smart-transaction-list-item.component.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch } from 'react-redux';
+import ListItem from '../../ui/list-item';
+import TransactionStatus from '../transaction-status/transaction-status.component';
+import TransactionIcon from '../transaction-icon';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { formatDateWithYearContext } from '../../../helpers/utils/util';
+import {
+  TRANSACTION_GROUP_CATEGORIES,
+  TRANSACTION_GROUP_STATUSES,
+} from '../../../../shared/constants/transaction';
+
+import CancelButton from '../cancel-button';
+import { cancelSwapsSmartTransaction } from '../../../ducks/swaps/swaps';
+
+export default function SmartTransactionListItem({
+  smartTransaction,
+  isEarliestNonce = false,
+}) {
+  const dispatch = useDispatch();
+  const t = useI18nContext();
+  const [cancelSwapLinkClicked, setCancelSwapLinkClicked] = useState(false);
+  const {
+    sourceTokenSymbol,
+    destinationTokenSymbol,
+    time,
+    status,
+  } = smartTransaction;
+  const category = TRANSACTION_GROUP_CATEGORIES.SWAP;
+  const title = t('swapTokenToToken', [
+    sourceTokenSymbol,
+    destinationTokenSymbol,
+  ]);
+  const subtitle = 'metamask';
+  const date = formatDateWithYearContext(time);
+  let displayedStatusKey;
+  if (status === 'pending') {
```

Comment:
Can we just use `TRANSACTION_GROUP_STATUSES.PENDING` here?

Reply:
Yea, we could switch it over to `TRANSACTION_GROUP_STATUSES.PENDING`. I have been hesitant because 'pending' reflects an enum from the smart transactions controller and `TRANSACTION_GROUP_STATUSES.PENDING` is a constant that reflects the status of normal transactions. So if one of those values changes, this piece of code will break. But I think if they are unlikely to change over time, it should be fine to change it.

Reply:
If we would be using a constant, I would prefer to have a new one, which is specific to Smart Transactions. E.g. `STX_STATUSES` or `SMART_TRANSACTION_STATUSES`.